src/app/shared/menu/menu.service.ts

```diff
@@ -1,24 +1,35 @@
 import { Injectable } from '@angular/core';
-import { MemoizedSelector, select, Store } from '@ngrx/store';
-import { MenuSection, MenuSectionIndex, MenuSections, MenusState, MenuState } from './menu.reducer';
+import { createSelector, MemoizedSelector, select, Store } from '@ngrx/store';
+import { MenuSection, MenuSections, MenuState } from './menu.reducer';
 import { AppState, keySelector } from '../../app.reducer';
 import { MenuID } from './initial-menus-state';
-import { Observable } from 'rxjs';
+import { combineLatest as observableCombineLatest, Observable } from 'rxjs';
 import { map, switchMap } from 'rxjs/operators';
 import {
   ActivateMenuSectionAction,
   AddMenuSectionAction,
-  CollapseMenuAction, CollapseMenuPreviewAction,
+  CollapseMenuAction,
+  CollapseMenuPreviewAction,
   DeactivateMenuSectionAction,
-  ExpandMenuAction, ExpandMenuPreviewAction,
+  ExpandMenuAction,
+  ExpandMenuPreviewAction,
   HideMenuAction,
   RemoveMenuSectionAction,
   ShowMenuAction,
   ToggleActiveMenuSectionAction,
   ToggleMenuAction,
 } from './menu.actions';
-import { hasNoValue, isNotEmpty } from '../empty.util';
-import { combineLatest as observableCombineLatest } from 'rxjs';
+import { hasNoValue, hasValue, isNotEmpty } from '../empty.util';
+
+export function menuKeySelector<T>(key: string, selector): MemoizedSelector<MenuState, T> {
```

Comment:
This new function needs TypeDocs.  Also, according to Coveralls, this is one of the areas where code coverage is dropping...this new method has no corresponding tests.  If you add tests, I suspect we'll get into a reasonable code coverage range.